### ISSUE TITLE
How to get loss of adapter-model?

### ISSUE THREAD

**Issue by songmh99** [adapter-hub/adapter-transformers#541]
## Environment info

     
- `adapter-transformers` version:
- Platform:
- Python version:
- PyTorch version (GPU?):

## Details

I want to use knowledge distillation to between two adapter models.
However, how can i get the student loss in DistillationTrainer? I can't get the labels from inputs :(

`  
class DistillationTrainer(Trainer):
    def __init__(self, *args, teacher_model=None, **kwargs):
        super().__init__(*args, **kwargs)
        self.teacher = teacher_model
        # place teacher on same device as student
        self._move_model_to_device(self.teacher,self.model.device)
        self.teacher.eval()
        self.labels = self.train_dataset['labels']
        
    def compute_loss(self, model, inputs, return_outputs=False):
        print('inputs:   ',inputs)
        # compute student output
        outputs_student = model(**inputs)
        
        student_loss=outputs_student.loss
        print(type(student_loss))
        if type(student_loss)!='float':
            student_loss = loss_f(outputs_student.logits.view(-1, self.model.config.num_labels),self.labels.view(-1))
        # compute teacher output
        with torch.no_grad():
            outputs_teacher = self.teacher(**inputs)

        # assert size
        assert outputs_student.logits.size() == outputs_teacher.logits.size()

        # Soften probabilities and compute distillation loss
        loss_function = nn.KLDivLoss(reduction="batchmean")
        loss_logits = (loss_function(
            F.log_softmax(outputs_student.logits / self.args.temperature, dim=-1),
            F.softmax(outputs_teacher.logits / self.args.temperature, dim=-1)) * (self.args.temperature ** 2))
        # Return weighted student loss
        loss = self.args.alpha * student_loss + (1. - self.args.alpha) * loss_logits
        return (loss, outputs_student) if return_outputs else loss
`

I print the "inputs" and found that: when the student model is an adapter model, the "inputs" don't have "labels"

**Comment by hSterz** [comment]
Hey @songmh99, have you tried setting `remove_unused_columns` to `False`in the `TrainerArgumets`? Because of some slight differences in the arguments of the forward pass compared to the huggingface transformers models, this can sometimes cause problems.

**Comment by songmh99** [comment]
> Hey @songmh99, have you tried setting `remove_unused_columns` to `False`in the `TrainerArgumets`? Because of some slight differences in the arguments of the forward pass compared to the huggingface transformers models, this can sometimes cause problems.

Thanks a lot, it does work!   :)
